pydra/engine/audit.py

```diff
@@ -173,15 +174,21 @@ def audit_task(self, task):
         import subprocess as sp
 
         label = task.name
-        if hasattr(task.inputs, "executable"):
-            command = task.cmdline
-        # assume function task
-        else:
-            command = None
-        # if hasattr(task.inputs, "in_file"):
-        #     input_file = task.inputs.in_file
-        # else:
-        #     input_file = None
+        entity_label = type(label)
+
+        command = task.cmdline if hasattr(task.inputs, "executable") else None
+
+        attr_list = attr_fields(task.inputs)
+        for attrs in attr_list:
+            if attrs.type in [File, Directory]:
+                input_name = attrs.name
```

Comment:
it will work well if only one element is either `File` or `Directory`, but if you have more than one element you will be overwriting `input_name`, `input_path`, etc. You have to store these variables for each element.

Reply:
Thanks for the response. I'm a bit confused. I attempted to implement a list of File/Directories and then iterate the list but it didn't work as expected.

Reply:
Creating a list to iterate over later would be fine, but I don't see where you create the list you're writing about. You should probably create one before l.182, but I don't see any...